daemon/src/order/maker.rs

```diff
@@ -0,0 +1,383 @@
+
+pub struct Actor {
+    executor: command::Executor,
+    oracle_pk: XOnlyPublicKey,
+    get_announcement:
+        MessageChannel<oracle::GetAnnouncement, Result<olivia::Announcement, NoAnnouncement>>,
+    build_party_params: MessageChannel<wallet::BuildPartyParams, Result<PartyParams>>,
+    sign: MessageChannel<wallet::Sign, Result<PartiallySignedTransaction>>,
+    projection: xtra::Address<projection::Actor>,
+    n_payouts: usize,
+    decision_senders: HashMap<OrderId, oneshot::Sender<protocol::Decision>>,
+    db: sqlite_db::Connection,
+    latest_offers: MessageChannel<xtra_libp2p_offer::maker::GetLatestOffers, Option<MakerOffers>>,
+}
+
+impl Actor {
+    pub fn new(
+        n_payouts: usize,
+        oracle_pk: XOnlyPublicKey,
+        get_announcement: MessageChannel<
+            oracle::GetAnnouncement,
+            Result<olivia::Announcement, NoAnnouncement>,
+        >,
+        (db, process_manager): (sqlite_db::Connection, xtra::Address<process_manager::Actor>),
+        (build_party_params, sign): (
+            MessageChannel<wallet::BuildPartyParams, Result<PartyParams>>,
+            MessageChannel<wallet::Sign, Result<PartiallySignedTransaction>>,
+        ),
+        projection: xtra::Address<projection::Actor>,
+        latest_offers: MessageChannel<
+            xtra_libp2p_offer::maker::GetLatestOffers,
+            Option<MakerOffers>,
+        >,
+    ) -> Self {
+        Self {
+            executor: command::Executor::new(db.clone(), process_manager),
+            oracle_pk,
+            get_announcement,
+            build_party_params,
+            sign,
+            projection,
+            n_payouts,
+            decision_senders: HashMap::default(),
+            db,
+            latest_offers,
+        }
+    }
+
+    async fn await_order(
+        &mut self,
+        framed: &mut Framed<Substream, JsonCodec<MakerMessage, TakerMessage>>,
+    ) -> Result<TakerMessage> {
+        let order = framed
+            .next()
+            .timeout(ORDER_TIMEOUT, || tracing::debug_span!("receive order"))
+            .await
+            .context("Timout when waiting for order")?
+            .context("Stream terminated")?
+            .context("Unable to decode order")?;
+
+        Ok(order)
+    }
+
+    async fn pick_offer(&self, offer_id: OfferId) -> Result<Offer> {
+        // TODO: Evaluate if this can be a problem during sending new offers to all connected peers
+        // -> Should be fine because the offer is already outdated by then and we should actually
+        // wait until finished?
+        // -> Cannot use send_async_next here - is that expected?
+        let latest_offers = match self
+            .latest_offers
+            .send(xtra_libp2p_offer::maker::GetLatestOffers)
+            .await
+        {
+            Ok(latest_offers) => latest_offers,
+            Err(e) => {
+                bail!("Failed to retrieve latest offer from offers actor: {e:#}")
+            }
+        };
```

Comment:
🔧 Using `anyhow::Context` is good here.

```suggestion
        let latest_offers = match self
            .latest_offers
            .send(xtra_libp2p_offer::maker::GetLatestOffers)
            .await
            .context("Failed to retrieve latest offer from offers actor")?;
```

And the same goes for the two `match`es below.

Reply:
😅 I wonder what my brain was thinking when I wrote this